File: MATTHEW/leetcode/AIcoding/5.Qwen-3/erlang_solutions/1513-number-of-substrings-with-only-1s.erl
-module(solution).
-export([num_substrings/1]).

num_substrings(S) ->
    num_substrings(S, 0, 0).

num_substrings([], _, Acc) ->
    Acc;
num_substrings([C | T], Count, Acc) ->
    if
        C == $1 ->
            NewCount = Count + 1,
            num_substrings(T, NewCount, Acc + NewCount);
        true ->
            num_substrings(T, 0, Acc)
    end.

File: leetcode/AIcoding/5.Qwen-3/python3_solutions/1513-number-of-substrings-with-only-1s.py
class Solution:
    def numSubstrings(self, s: str) -> int:
        count = 0
        n = len(s)
        for i in range(n):
            if s[i] == '1':
                j = i
                while j < n and s[j] == '1':
                    count += 1
                    j += 1
        return count